### PR TITLE
Update: Implement the new discussion panel design.

### PR DESCRIPTION
**Pull request by jorgefilipecosta**
Implements the new discussion panel design proposed at: https://github.com/WordPress/gutenberg/issues/59689#issuecomment-2090763388

cc: @jameskoster 

## Screenshots

Screenshot 2024-05-03 at 13 48 35

## Testing Instructions

Tested the discussion panel options and verified they work as expected and are correctly saved.

### CODE REVIEW

**Comment by jameskoster** [comment]
Thanks for the ping.

The radio controls seem to be a bit squished:

Screenshot 2024-05-07 at 10 48 14

We should probably fix that in the component (cc @WordPress/gutenberg-components - there seems to be a missing `flex-shrink`), but for this PR it can be addressed by adding a 320px min-width to the popover. This will align with the Status dialog and improve the readability.

Speaking of aligning with the Status dialog, would it be possible to match the spacing between radios?

---

The button label references pingbacks and trackbacks when editing pages, but as I understand it pages don't support them?

---

We need to do something about this button label:

Screenshot 2024-05-07 at 10 53 28

Maybe something like:

* Comments & Pingbacks open: "Discussion: Open"
* Comments open, pingbacks not allowed: "Discussion: Comments only"
* Comments closed, pingbacks allowed: "Discussion: Pings only"
* Comments & pingbacks closed: "Discussion: Closed"

**Comment by tyxla** [comment]
> The radio controls seem to be a bit squished:

> We should probably fix that in the component (cc @WordPress/gutenberg-components - there seems to be a missing `flex-shrink`), but for this PR it can be addressed by adding a 320px min-width to the popover. This will align with the Status dialog and improve the readability.

This is a valid issue and it will occur with the `RadioControl` in other instances, so I think we should fix it properly. I'd recommend the broader fix rather than the `min-width` hack. 

I've worked on resolving this independently in #61476. I think `flex-shrink` won't work because we really want constant width here. @jorgefilipecosta @jameskoster let me know what you think.

**Comment by jameskoster** [comment]
@tyxla Thanks, I'll take a look at your PR.

To be clear I think we should do both;

1. Fix the issue upstream in the component.
2. Apply the min-width to this particular panel – not just because it temporarily fixes the radio size, but also because it will then be consistent with other panels in this UI, and because it improves the legibility of the text.

**Comment by jorgefilipecosta** [comment]
Hi @jameskoster I applied the min-width to the popover and used the suggested labels. Thank you for the review and feedback let me know if there are any other updates we should do.

**Comment by jameskoster** [comment]
Thanks @jorgefilipecosta, one small tweak;

Since pages do not support pingbacks, the button label can just be 'Open' or 'Closed'. Currently if you set discussion to open while editing a page the button says 'Comments only' which might suggest there are other options.

Otherwise this looks good :)

**Comment by jorgefilipecosta** [comment]
> Thanks @jorgefilipecosta, one small tweak;
> 
> Since pages do not support pingbacks, the button label can just be 'Open' or 'Closed'. Currently if you set discussion to open while editing a page the button says 'Comments only' which might suggest there are other options.
> 
> Otherwise this looks good :)
 
Hi @jameskoster, I applied the suggestion and made the labels dependent on the post type support for trackbacks and comments. Let me know if it has the green light now :)

**Review by jameskoster — APPROVED** [review]
I pushed a small tweak to adjust spacing, and make the documentation link help text for the checkbox control which seems semantically appropriate.

The mis-alignment of text between radios/checkboxes is bothersome, but that's something to fix at the component level.

**Comment by Mamaduka** [comment]
Every time I open the new panel, I noticed that Chrome logs the following error.

## Screenshot
CleanShot 2024-05-22 at 09 59 19

File: test/e2e/specs/editor/various/sidebar.spec.js [hunk 1]
```diff
@@ -123,13 +122,17 @@ test.describe( 'Sidebar', () => {
...
+			name: 'Change discussion options',
+		} );
 		const postSummarySection = page.getByRole( 'checkbox', {
 			name: 'Stick to the top of the blog',
 		} );
 
 		await expect( postExcerptPanel ).toBeVisible();
 		await expect( postFeaturedImagePanel ).toBeVisible();
 		await expect( postSummarySection ).toBeVisible();
+		await expect( postDiscussionPanel ).toHaveCount( 1 );
```
Comment (Mamaduka) [inline test/e2e/specs/editor/various/sidebar.spec.js:135]: It's recommended to use explicit assertions. See - https://github.com/WordPress/gutenberg/blob/trunk/docs/contributors/code/e2e/README.md#make-explicit-assertions.

File: test/e2e/specs/editor/various/sidebar.spec.js [hunk 2]
```diff
@@ -147,5 +150,6 @@ test.describe( 'Sidebar', () => {
 		await expect( postExcerptPanel ).toBeHidden();
 		await expect( postFeaturedImagePanel ).toBeHidden();
 		await expect( postSummarySection ).toBeHidden();
+		await expect( postDiscussionPanel ).toHaveCount( 0 );
```
Comment (Mamaduka) [inline test/e2e/specs/editor/various/sidebar.spec.js:153]: Same here.

**Pull request merged**